activerecord/lib/arel/nodes/and.rb

```diff
@@ -18,6 +18,12 @@ def right
         children[1]
       end
 
+      def fetch_attribute(&block)
+        children.each do |child|
```

Comment:
I looked over the code
As you said, fetch_attribute should return a boolean value

fetch_attribute seems to be a method that determines whether the return value of the block is true for all attributes it owns

So it seems correct to use `all?` instead of `each` this time. What do you think?